src/cli/subcommands/car_cmd.rs

```diff
@@ -204,21 +193,144 @@ mod tests {
+        let output = tempfile::Builder::new()
+            .prefix("forest-cli-car-concat-same-file")
+            .suffix(FOREST_CAR_ZST_SUFFIX)
+            .tempfile()?;
+
+        cli()?
+            .arg("car")
+            .arg("concat")
+            .arg("./test-snapshots/chain4.car")
+            .arg("./test-snapshots/chain4.car")
+            .arg("-o")
+            .arg(output.path().as_os_str().to_str().unwrap())
+            .assert()
+            .success();
+
+        validate_car(output.path()).await?;
+
+        Ok(())
+    }
+
+    #[tokio::test]
+    async fn forest_cli_car_concat_same_file_3_times() -> Result<()> {
+        let output = tempfile::Builder::new()
+            .prefix("forest-cli-car-concat-same-file-3-times")
+            .suffix(FOREST_CAR_ZST_SUFFIX)
+            .tempfile()?;
+
+        cli()?
+            .arg("car")
+            .arg("concat")
+            .arg("./test-snapshots/chain4.car")
+            .arg("./test-snapshots/chain4.car")
+            .arg("./test-snapshots/chain4.car")
+            .arg("-o")
+            .arg(output.path().as_os_str().to_str().unwrap())
+            .assert()
+            .success();
+
+        validate_car(output.path()).await?;
+
+        Ok(())
+    }
+
+    async fn new_car(size: usize, path: impl AsRef<Path>) -> Result<()> {
+        let rng = SmallRng::seed_from_u64(0xdeadbeef);
+        let root_block = new_block(&mut rng.clone());
+        let roots = vec![root_block.cid];
+        let block_stream = futures::stream::unfold(rng, |mut rng| async {
+            Some((Ok(new_block(&mut rng)), rng))
+        })
+        .take(size);
+        let frames = crate::db::car::forest::Encoder::compress_stream(
+            8000_usize.next_power_of_two(),
+            zstd::DEFAULT_COMPRESSION_LEVEL as _,
+            block_stream,
+        );
+        let mut writer = tokio::io::BufWriter::new(tokio::fs::File::create(path).await?);
+        crate::db::car::forest::Encoder::write(&mut writer, roots, frames).await?;
+        writer.flush().await?;
+
+        Ok(())
+    }
+
+    fn new_block(rng: &mut SmallRng) -> Block {
+        let mut data = [0; 64];
+        rng.fill(&mut data);
+        let cid = Cid::new_v1(DAG_CBOR, multihash::Code::Blake2b256.digest(&data));
+        Block {
+            cid,
+            data: data.to_vec(),
+        }
+    }
+
+    async fn validate_car(path: impl AsRef<Path>) -> Result<()> {
+        let reader = CarStream::new(tokio::io::BufReader::new(
+            tokio::fs::File::open(path).await?,
+        ))
+        .await?;
+        assert!(!reader.header.roots.is_empty());
+        let mut count = 0;
+        let blocks: Vec<_> = reader.inspect_ok(|_| count += 1).try_collect().await?;
+        assert!(blocks.iter().map(|b| b.cid).all_unique());
+        println!("Result car block count: {count}");
+        Ok(())
+    }
+
+    fn cli() -> Result<Command> {
+        Ok(Command::cargo_bin("forest-cli")?)
```

Comment:
`cargo_bin` only works from the context of integration tests.

Reply:
Does it? I can see these tests passing locally and on CI

Reply:
It says so in the documentation: https://docs.rs/assert_cmd/latest/assert_cmd/cargo/index.html

Could you verify that it works even if the binaries aren't built first?

Reply:
`cargo test` builds binaries by default but if `--lib` flag is used it won't build binaries. We have a few options
1. Delete these tests.
2. Move back to tests/ and make `util` and `db` mod public
3. build `forest-cli` with `escargot` (but it takes more than 30s and will timeout)

Reply:
Removed